### PR TITLE
Adds the Wood Datum Material, giving it to items made from wood.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

We have plenty of wooden items on-station at the moment anyway, so there's no real issue with expanding datum materials for these. I don't know if biomass was originally supposed to be wood or not, but I imagine that's supposed to be more like "plant matter" than anything else, so I went with this.

Anything made out of wood automatically becomes flamable, and by default wooden materials have weaker melee damage if it's affected by datum mats, to work more in line with being ["fake weapons"](https://www.youtube.com/watch?v=x5-JVvCrGC8).

## Why It's Good For The Game

Expands Datum Mats, making wood recyclable to a degree, and makes wood behave more like "wood".

## Changelog
:cl:
tweak: Wood is now a datum material, and behaves like other datum materials.
/:cl:

### CODE REVIEW

**Comment by knacker48** [comment]
Have you adjusted existing weapons made of wood to do the same damage as before or are you nerfing them across the board?

**Comment by ATH1909** [comment]
This won't affect the base damage of baseball bats, right?

**Comment by ArcaneMusic** [comment]
> Have you adjusted existing weapons made of wood to do the same damage as before or are you nerfing them across the board?



> This won't affect the force(s) of baseball bats, right?

Items made from datum mats specifically are the only ones that are affect by stat changes related to the datum mat's status. So, baseball bats are exempt, but things like a wooden toolbox or the cleric's mace aren't and have their stats based on the material itself.

**Comment by necromanceranne** [comment]
I actually don't see the logic in a wooden mace being a weak melee weapon when it's a fucking club. A wooden weapon applied correctly, shaped properly and using the right kinds of wood is as much a deadly weapon as one made of metal. The only significant difference is how heavily protected the person you are smacking is from attacks.

Tl;dr blunt force weapons are fucking deadly as shit and a dense enough wood will hurt a lot. It would be more accurate if it had negative armor penetration.

**Comment by nemvar** [comment]
I think biomass is supposed to be the thing in the botany production thing.

File: code/datums/components/crafting/guncrafting.dm [hunk 1]
```diff
@@ -11,5 +11,6 @@
 /obj/item/weaponcrafting/stock
 	name = "rifle stock"
 	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood."
+	custom_materials = list(/datum/material/wood = 12000)
```
Comment (Qustinnus) [inline code/datums/components/crafting/guncrafting.dm]: thats a lot of wood

**Comment by ATH1909** [comment]
> I actually don't see the logic in a wooden mace being a weak melee weapon when it's a fucking club. A wooden weapon applied correctly, shaped properly and using the right kinds of wood is as much a deadly weapon as one made of metal. The only significant difference is how heavily protected the person you are smacking is from attacks.
> 
> Tl;dr blunt force weapons are fucking deadly as shit and a dense enough wood will hurt a lot. It would be more accurate if it had negative armor penetration.

Maybe only make wooden weapons do less damage if they're sharp?

**Comment by Krysonism** [comment]
> 
> 
> > Have you adjusted existing weapons made of wood to do the same damage as before or are you nerfing them across the board?
> 
> > This won't affect the force(s) of baseball bats, right?
> 
> Items made from datum mats specifically are the only ones that are affect by stat changes related to the datum mat's status. So, baseball bats are exempt, but things like a wooden toolbox or the cleric's mace aren't and have their stats based on the material itself.

This is not true, at least it didn't use to be true. When i made my amputation shears i had to lower force since the titanium content increased it by a lot.

File: code/datums/components/crafting/guncrafting.dm (hunk 1 above)
Comment (ArcaneMusic) [reply code/datums/components/crafting/guncrafting.dm]: It's made from 10 planks of wood originally, you lose 8k in the carving process.

**Comment by ArcaneMusic** [comment]
> > > Have you adjusted existing weapons made of wood to do the same damage as before or are you nerfing them across the board?
> > 
> > 
> > > This won't affect the force(s) of baseball bats, right?
> > 
> > 
> > Items made from datum mats specifically are the only ones that are affect by stat changes related to the datum mat's status. So, baseball bats are exempt, but things like a wooden toolbox or the cleric's mace aren't and have their stats based on the material itself.
> 
> This is not true, at least it didn't use to be true. When i made my amputation shears i had to lower force since the titanium content increased it by a lot.

Well, in any case, I've double checked to confirm that the force is still 10, both as a debug spawn and stack crafted item.

**Comment by ExcessiveUseOfCobblestone** [comment]
if the object has `MATERIAL_AFFECT_STATISTICS` in the materials flags it will touch force and the like.

in this PR though, ALL objects regardless if they have this flag or not will become burnable if it has wood in it. I'll be reviewing this tonight.

File: code/datums/materials/basemats.dm [hunk 2]
```diff
@@ -189,6 +189,29 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+	categories = list(MAT_CATEGORY_RIGID = TRUE)
+	value_per_unit = 0.06
+	beauty_modifier = 0.1
+	armor_modifiers = list("melee" = 1.1, "bullet" = 1.1, "laser" = 0.4, "energy" = 0.4, "bomb" = 1, "bio" = 0.2, "rad" = 0, "fire" = 0, "acid" = 0.3)
+
+/datum/material/wood/on_applied_obj(atom/source, amount, material_flags)
+	. = ..()
+	if(istype(source, /obj/item))
+		var/obj/item/wooden = source
+		wooden.resistance_flags |= FLAMMABLE
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]: this should have the stat flag requirement too.

File: code/datums/materials/basemats.dm [hunk 3]
```diff
@@ -189,6 +189,29 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+
+/datum/material/wood/on_removed_obj(atom/source, material_flags)
+	. = ..()
+	if(istype(source, /obj/item))
+		var/obj/item/wooden = source
+		wooden.resistance_flags &= ~FLAMMABLE
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]: same

File: code/datums/materials/basemats.dm [hunk 4]
```diff
@@ -189,6 +189,29 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+	id = "wood"
+	desc = "Flexible, durable, but flamable. Hard to come across in space."
+	color = "#bb8e53"
+	strength_modifier = 0.5
+	categories = list(MAT_CATEGORY_RIGID = TRUE)
+	value_per_unit = 0.06
+	beauty_modifier = 0.1
+	armor_modifiers = list("melee" = 1.1, "bullet" = 1.1, "laser" = 0.4, "energy" = 0.4, "bomb" = 1, "bio" = 0.2, "rad" = 0, "fire" = 0, "acid" = 0.3)
+
+/datum/material/wood/on_applied_obj(atom/source, amount, material_flags)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]:
```suggestion
/datum/material/wood/on_applied_obj(obj/source, amount, material_flags)
```

This is the object proc so you know it's an object already. I personally would remove the istype tbh, any obj should become flammable if it's made of wood 😏

File: code/datums/materials/basemats.dm [hunk 5]
```diff
@@ -189,6 +189,29 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+	. = ..()
+	if(istype(source, /obj/item))
+		var/obj/item/wooden = source
+		wooden.resistance_flags |= FLAMMABLE
+
+/datum/material/wood/on_removed_obj(atom/source, material_flags)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]:
```suggestion
/datum/material/wood/on_removed_obj(obj/source, material_flags)
```

File: code/game/objects/items/cigs_lighters.dm [hunk 6]
```diff
@@ -101,6 +101,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	name = "firebrand"
 	desc = "An unlit firebrand. It makes you wonder why it's not just called a stick."
 	smoketime = 20 //40 seconds
+	custom_materials = list(/datum/material/wood = 2000)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/cigs_lighters.dm]: can you base all these values off the MINERAL_MATERIAL_AMOUNT? that sounds better than magic numbering something that's related to a define.

File: code/datums/materials/basemats.dm (hunk 2 above)
Comment (ArcaneMusic) [reply code/datums/materials/basemats.dm]: Run this one by me one more time when you have a second, I haven't worked with bitflags before this one.

Comment (ExcessiveUseOfCobblestone) [reply code/datums/materials/basemats.dm]: https://github.com/tgstation/tgstation/blob/e87033d792cf4834f4fc18b0e6a2c2057859ce89/code/datums/materials/_material.dm#L50-L55

File: code/datums/materials/basemats.dm [hunk 7]
```diff
@@ -189,6 +189,27 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+	color = "#bb8e53"
+	strength_modifier = 0.5
+	categories = list(MAT_CATEGORY_RIGID = TRUE)
+	value_per_unit = 0.06
+	beauty_modifier = 0.1
+	armor_modifiers = list("melee" = 1.1, "bullet" = 1.1, "laser" = 0.4, "energy" = 0.4, "bomb" = 1, "bio" = 0.2, "rad" = 0, "fire" = 0, "acid" = 0.3)
+
+/datum/material/wood/on_applied_obj(obj/source, amount, material_flags)
+	. = ..()
+	var/obj/item/wooden = source
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]:
```suggestion
	var/obj/wooden = source
```

File: code/datums/materials/basemats.dm [hunk 8]
```diff
@@ -189,6 +189,27 @@ Unless you know what you're doing, only use the first three numbers. They're in
...
+	wooden.resistance_flags |= FLAMMABLE
+
+/datum/material/wood/on_removed_obj(obj/source, material_flags)
+	. = ..()
+	var/obj/item/wooden = source
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/materials/basemats.dm]:
```suggestion
	var/obj/wooden = source
```
